libminifi/include/core/ConfigurableComponent.h

```diff
@@ -56,6 +56,9 @@ class ConfigurableComponent {
   ConfigurableComponent& operator=(const ConfigurableComponent &other) = delete;
   ConfigurableComponent& operator=(ConfigurableComponent &&other) = delete;
 
+  template<typename T = std::string, typename = std::enable_if_t<std::is_default_constructible_v<T>>>
+  std::optional<T> getProperty(const Property& property) const;
```

Comment:
why is this in this PR?

Reply:
I tried to avoid opening a PR for these lines, should I remove this?

Reply:
what is it for?
you could add a commit with an explanation of this change, so whoever will merge the PR can include it in the squashed commit message

Reply:
added commit message

Reply:
Please use `requires` instead of `enable_if`. We can't yet use standard library concepts (like [`std::default_initializable`](https://en.cppreference.com/w/cpp/concepts/default_initializable)) because of xcode, but we can use requires declarations with type traits.

And I think T should be movable, too, so that we can move the value to the result object.

Reply:
removed it